sql/_functions.sql

```diff
@@ -23,6 +23,6 @@ $$ LANGUAGE sql;
 CREATE OR REPLACE FUNCTION get_boro(_geom geometry) 
 RETURNS varchar AS $$
     SELECT b.borocode::varchar
-    FROM dcp_boroboundaries_wi b
-    WHERE ST_Within(_geom, b.wkb_geometry)
+    FROM dcp_boroboundaries b
+    WHERE ST_Intersects(_geom, b.wkb_geometry)
```

Comment:
Do we have cases in this dataset where geoms might intersect more than one boro? Do you know what will happen in that case with this function - will it return a single varchar as the type signature suggests (first boro it finds?), or will it throw an error?

Also, a bit unsure if this really should be a function - it's basically just left joining dcp_boroboundaries to the geom column that this function is being applied to, I would imagine just doing that join would be more performant (as long as there's an index on the geom column). It's also only called in `create_kpdb`, so I would lean with that personally

Reply:
Actually didn't know the answer to my first question, so did some testing locally - 
``` sql
drop if exists table "test";
create temp table "test" ("a", "b") as values (1, 2), (1, 1);
select * from test;
create or replace function test_function(val int)
returns int as $$
    select b from test where a=val
    $$ language sql;

select test_function(a), b from "test";
```

returns 
| test_function | b |
| -- | -- |
| 2 | 2 |
| 2 | 1 |

So seems to just be grabbing one of the values where it matches, in this case the "last" though I don't know how much to read into that.

Reply:
Not saying second is ideal, but at least then we can actually tell when we're intersecting multiple boros. The first we're potentially somewhat randomly (or at least indiscriminately) losing information in these cases

Reply:
I don't think so. The project polygons look to be exclusively in the regular borough boundaries but this was an issue when we were joining on the `dcp_boroboundaries_wi` water included boundarie of Manhattan borders the land border of brooklyn so all of the piers that are being redeveloped on the BK waterfront were being misclassified as Manhattan records because they intersected both BK and Manhattan. Previously the KPDB pipeline was bringing in the water included file and then using st_within which was creating issues because the projects weren't always totally within the borough boundary file. 

Yeah I think from a performance side its probably best to drop the function and just do a join but I think maybe we can do that when we refactor KPDB, I'm overly cautious changing anything at this point and just want it to be done. I also think that we might need to make some adjustments to how we assign boroughs for records without a geometry which hopefully will be ironed out when we meet at 2pm today

Reply:
So tried running both ways

Current method, ~9 min to query on my machine
27989	4
23815	3
18306	5
9731	2
4552	1

Add index, left join dcp_boroboundaries, ~4 min to query on my machine
27989	4
23823	3
18306	5
9731	2
4552	1

So looks like we have 8 records on the brooklyn queens border that intersect both, all of which are getting assigned to queens. I'll check these out after lunch

Reply:
Following records
| source | record_id | record_name |
|-|-|-|
|DCP Application | 2021K0237 | River Ring|
|DCP Application | 2021K0313 | 1 Java Street Waterfront|
|DCP Application | 2023K0138 | Herkimer - Williams - City Map Change|
|DCP Application | P2014K0254 | Williamsburg Edge Phase 2 Waterfront Certification|
|DOB | 320377463 | 130 FURMAN STREET|
|DOB | 320513182 | 90 FURMAN STREET|
|DOB | B00539478 | 18 INDIA STREET|
|DOB | B00629573 | 105 RIVER STREET|